src/main/org/bson/ByteBuf.java

```diff
@@ -488,6 +506,34 @@ public ByteBuf putInt32(ThreadContext context, IRubyObject value) {
     return this;
   }
 
+  /**
+   * Put an unsigned 32 bit integer onto the buffer.
+   *
+   * @param value The integer to write.
+   * 
+   */
+  @JRubyMethod(name = "put_uint32")
+  public ByteBuf putUInt32(ThreadContext context, IRubyObject value) {
+    if (value instanceof RubyFloat) {
+      throw getRuntime().newArgumentError("put_uint32; incorrect type: float, expected: integer");
+    }
+    ensureBsonWrite(4);
+
+    long temp = RubyNumeric.fix2long((RubyFixnum) value);
+    
+    if (temp >= Math.pow(2, 32) || temp < 0) {
+      throw getRuntime().newRangeError(format("Number %d is out of range [0, 2^32)", temp));
```

Comment:
Is there a reason to provide different error messages for the java extensions and C extensions? I think that could be potentially confusing to both users and developers.